src/auth/models.cpp

```diff
@@ -471,12 +474,21 @@ Role Role::Deserialize(const nlohmann::json &data) {
   auto permissions = Permissions::Deserialize(data[kPermissions]);
 #ifdef MG_ENTERPRISE
   if (memgraph::license::global_license_checker.IsEnterpriseValidFast()) {
+    Databases db_access;
+    if (data[kDatabases].is_structured()) {
+      db_access = Databases::Deserialize(data[kDatabases]);
+    } else {
+      // Back-compatibility
+      spdlog::warn("Role without specified database access. Given access to the default database.");
+      db_access.Add(dbms::kDefaultDB);
+      db_access.SetDefault(dbms::kDefaultDB);
```

Comment:
Maybe do this as an auth migration step, rather than perpetual backwards compatibility

Reply:
There are 3 scenarios that we need to cover and I'm not sure migration is the solution here.
1) community -> does not generate the database kv element
2) enterprise no license -> generates it, but leaves it empty
3) enterprise with license -> generates it and actually stores the data